core/bin/zksync_server/src/main.rs

```diff
@@ -156,10 +156,7 @@ async fn main() -> anyhow::Result<()> {
 
     // OneShotWitnessGenerator is the only component that is not expected to run indefinitely
     // if this value is `false`, we expect all components to run indefinitely: we panic if any component returns.
-    let is_only_oneshot_witness_generator_task = matches!(
-        components.as_slice(),
-        [Component::WitnessGenerator(Some(_), _)]
-    );
+    let is_only_oneshot_witness_generator_task = false;
```

Comment:
this var is passed further to the `initialize_components` function as `use_prometheus_push_gateway` argument. This is very unclear to me and I do not understand how they are connected.

Reply:
AFAIU, oneshot witgen is the only configuration that makes the server stop on its own after some time; i.e., it's not supposed to run infinitely as a server. That's why metrics are pushed to the gateway instead of being pulled. IMO, it'd make sense to simplify `initialize_components` and this function to "inline" this value / get rid of unnecessary args.